Assets/Scripts/UI/PopupSpammer.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections;
+using CollectionExtensions;
+using UnityEngine;
+
+public class PopupSpammer : MonoBehaviour
```

Comment:
Maybe a bit nitpicky but, I think this script might be placed better on the player prefab if it was attached to the HUD gameobject instead of the player root. I think we could give playerHud a `public PlayerHud => get;` field in playermanager and gain access to the PopupSpammer from the HUD gameobject instead of through the playerManager gameobject.

Reply:
Yeah, it did feel weird to put this component on the player :upside_down_face:

Reply:
Done!